code/modules/hivemind/core.dm

```diff
@@ -0,0 +1,59 @@
+//Hivemind - is rogue AI, that use unknown nanotech
+//Main goal of this machine is spread across the ship and annihilate so much stuff as he can
+//We are actually don't know who created this ai, how much time it wanders in space or, maybe, any other way?
+//We even don't know why this ai do it.
+
+
+var/datum/hivemind/hive_mind_ai
+
+/datum/hivemind
+	var/name
+	var/surname
+	var/evo_points = 0
+	var/evo_points_max = 1000
+	var/failure_chance = 55				//failure chance is lowers each 10 EP
+	var/list/hives = list() 			//all functional hives stored here
+	//i know, whitelist is bad, but it's required here
+	var/list/restricted_machineries = list( /obj/machinery/light,					/obj/machinery/atmospherics,
+											/obj/machinery/portable_atmospherics,	/obj/machinery/door,
+											/obj/machinery/camera,					/obj/machinery/light_switch,
+											/obj/machinery/disposal,				/obj/machinery/firealarm,
+											/obj/machinery/alarm,					/obj/machinery/recharger,
+											/obj/machinery/hologram,				/obj/machinery/holoposter,
+											/obj/machinery/button,					/obj/machinery/ai_status_display,
+											/obj/machinery/status_display,			/obj/machinery/requests_console,
+											/obj/machinery/newscaster,				/obj/machinery/floor_light,
+											/obj/machinery/nuclearbomb,				/obj/machinery/flasher,
+											/obj/machinery/filler_object)
+	//internals
+	var/list/global_abilities_cooldown = list()
+	var/list/EP_price_list = list()
+
+/datum/hivemind/New()
+	..()
+	name = pick("Reworker", "Executor", "Annihilator",
```

Comment:
Some literary feedback

Instead of Reworker, use Reclaimer and Shaper
Bugmerger just sounds silly, remove that one
Propagator instead of Reproducer
Creator instead of Maker